--- origsrc/kdelibs4support-5.12.0/src/kdecore/kstandarddirs.cpp	2015-07-05 03:57:10.000000000 -0500
+++ src/kdelibs4support-5.12.0/src/kdecore/kstandarddirs.cpp	2015-07-21 21:00:29.435038100 -0500
@@ -1714,7 +1714,7 @@ static QString readEnvPath(const char *e
     return QDir::fromNativeSeparators(QFile::decodeName(c_path));
 }
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__CYGWIN__)
 static QString executablePrefix()
 {
     char path_buffer[MAXPATHLEN + 1];
@@ -1784,7 +1784,7 @@ void KStandardDirs::addKDEDefaults()
     if (!execPrefix.isEmpty() && !kdedirList.contains(execPrefix, cs)) {
         kdedirList.append(execPrefix);
     }
-#ifdef __linux__
+#if defined(__linux__) || defined(__CYGWIN__)
     const QString linuxExecPrefix = executablePrefix();
     if (!linuxExecPrefix.isEmpty()) {
         kdedirList.append(linuxExecPrefix);
